### ISSUE TITLE
option to disable special handling of double-quotes in read_csv

### ISSUE THREAD

**Issue by cbilot** [pola-rs/polars#1528]
#### Describe your feature request
It would be helpful to have an option in polars.io.read_csv() to turn off the special handling of double-quotes as delimiters of embedded text.  That is, to treat double-quotes as just another character.

Double-quotes are (hopefully) inserted into csv files to delimit text fields that contain an embedded end-line (\n) or separator field (,).  For example: "Smith, Joe".  Or "It was a dark and stormy night.\n The train pulled into the station."

Too often, however, csv files contain double-quotes that are not delimiters of embedded text.  And the results can go wildly wrong, often with little warning.

#### A simple example
```python
import polars as pl
from io import StringIO

rolling_stones = (
"""
linenum,last_name,first_name
1,Jagger,Mick
2,O"Brian,Mary
3,Richards,Keith
4,L"Etoile,Bennet
5,Watts,Charlie
6,Smith,D"Shawn
7,Wyman,Bill
8,Woods,Ron
9,Jones,Brian
"""
)

my_csv_fl = StringIO(rolling_stones)
print(my_csv_fl.getvalue())
```
```
linenum,last_name,first_name
1,Jagger,Mick
2,O"Brian,Mary
3,Richards,Keith
4,L"Etoile,Bennet
5,Watts,Charlie
6,Smith,D"Shawn
7,Wyman,Bill
8,Woods,Ron
9,Jones,Brian
```

Now, let's parse this with polars.io.read_csv()
```python
my_df = pl.io.read_csv(file = my_csv_fl)
print(my_df)
```
```
┌─────────┬───────────┬────────────┐
│ linenum ┆ last_name ┆ first_name │
│ ---     ┆ ---       ┆ ---        │
│ i64     ┆ str       ┆ str        │
╞═════════╪═══════════╪════════════╡
│ 1       ┆ Jagger    ┆ Mick       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2       ┆ O"Brian   ┆ Mary       │
│         ┆           ┆ 3          │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 5       ┆ Watts     ┆ Charlie    │
└─────────┴───────────┴────────────┘
```
Notice how lines 2-4 are mangled.  But more adversely, all lines beyond line 5 are simply missing.  (This scenario happened to me, causing more than 10,000 lines at the end of a file to be silently skipped because the name L"Etoile appeared in a last name field.)

#### Discussion
The only reason I log this issue is that its an unrecoverable error.  That is, once the lines are skipped, I cannot later fix the issue in the resulting data frame that is returned by read_csv().

Heck, sometimes the only documentation accompanying large csv files is an email with "Here ya go".

#### Alternatives
Currently, I handle this by pre-processing each text file, turning each double-quote into a single-quote, and then running the processed file through read_csv().  Perhaps not ideal, given that Polars is targeted for those working with very large data sets and very large input files.

#### How others have handled this
In the "data.tables" package in R, there is a "**quote =**" keyword option on the **fread()** function to change the handling of quoted text.  And by setting **quote = ""**, I can turn off the special handling of double-quotes.

From the documentation of the **fread()** function in the data.table package:
```
Quotes:
When quote is a single character,
• Spaces and other whitespace (other than sep and \n) may appear in unquoted character fields,
e.g., ...,2,Joe Bloggs,3.14,....

• When character columns are quoted, they must start and end with that quoting character
immediately followed by sep or \n, e.g., ...,2,"Joe Bloggs",3.14,....

In essence quoting character fields are required only if sep or \n appears in the string value.

Quoting may be used to signify that numeric data should be read as text. Unescaped quotes
may be present in a quoted field, e.g., ...,2,"Joe,"Bloggs"",3.14,..., as well as escaped
quotes, e.g., ...,2,"Joe \",Bloggs\"",3.14,....

If an embedded quote is followed by the separator inside a quoted field, the embedded quotes
up to that point in that field must be balanced; e.g. ...,2,"www.blah?x="one",y="two"",3.14,....

On those fields that do not satisfy these conditions, e.g., fields with unbalanced quotes, fread
re-attempts that field as if it isn’t quoted. This is quite useful in reading files that contains
fields with unbalanced quotes as well, automatically.

To read fields as is instead, use quote = "".
```
Honestly, I wouldn't suggest going this far.  I'm sure that parsing csv files includes a lot of logic, conditions, and state handling.  (And I'm sure there are myriad other things that the Polars team would rather be doing.)

But perhaps a simple option to turn off the special-handling of double-quotes is sufficient for now.  Or perhaps a warning when the parser detects a problem.

**Comment by ghuls** [comment]
For wierd CSVs you can also try to fix them with `xsv input`
```
$ xsv input -h
Read CSV data with special quoting rules.

Generally, all xsv commands support basic options like specifying the delimiter
used in CSV data. This does not cover all possible types of CSV data. For
example, some CSV files don't use '"' for quotes or use different escaping
styles.

Usage:
    xsv input [options] [<input>]

input options:
    --quote <arg>          The quote character to use. [default: "]
    --escape <arg>         The escape character to use. When not specified,
                           quotes are escaped by doubling them.
    --no-quoting           Disable quoting completely.

Common options:
    -h, --help             Display this message
    -o, --output <file>    Write output to <file> instead of stdout.
    -d, --delimiter <arg>  The field delimiter for reading CSV data.
                           Must be a single character. (default: ,)


$ xsv input test_double_quotes.csv > test_double_quotes.fixed.csv

$ head test_double_quotes.csv 
linenum,last_name,first_name
1,Jagger,Mick
2,O"Brian,Mary
3,Richards,Keith
4,L"Etoile,Bennet
5,Watts,Charlie
6,Smith,D"Shawn
7,Wyman,Bill
8,Woods,Ron
9,Jones,Brian

$ head test_double_quotes.fixed.csv 
linenum,last_name,first_name
1,Jagger,Mick
2,"O""Brian",Mary
3,Richards,Keith
4,"L""Etoile",Bennet
5,Watts,Charlie
6,Smith,"D""Shawn"
7,Wyman,Bill
8,Woods,Ron
9,Jones,Brian
```

```
In [2]: pl.read_csv('test_double_quotes.fixed.csv')
Out[2]: 
shape: (9, 3)
┌─────────┬───────────┬────────────┐
│ linenum ┆ last_name ┆ first_name │
│ ---     ┆ ---       ┆ ---        │
│ i64     ┆ str       ┆ str        │
╞═════════╪═══════════╪════════════╡
│ 1       ┆ Jagger    ┆ Mick       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2       ┆ O"Brian   ┆ Mary       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3       ┆ Richards  ┆ Keith      │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4       ┆ L"Etoile  ┆ Bennet     │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...     ┆ ...       ┆ ...        │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 5       ┆ Watts     ┆ Charlie    │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6       ┆ Smith     ┆ D"Shawn    │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 7       ┆ Wyman     ┆ Bill       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 8       ┆ Woods     ┆ Ron        │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 9       ┆ Jones     ┆ Brian      │
└─────────┴───────────┴────────────┘

**Comment by ritchie46** [comment]
@cbilot with #1530 we get:

```python
import polars as pl

rolling_stones = """
linenum,last_name,first_name
1,Jagger,Mick
2,O"Brian,Mary
3,Richards,Keith
4,L"Etoile,Bennet
5,Watts,Charlie
6,Smith,D"Shawn
7,Wyman,Bill
8,Woods,Ron
9,Jones,Brian
"""

print(pl.read_csv(rolling_stones.encode(), quote_char=None))
```

```
shape: (9, 3)
┌─────────┬───────────┬────────────┐
│ linenum ┆ last_name ┆ first_name │
│ ---     ┆ ---       ┆ ---        │
│ i64     ┆ str       ┆ str        │
╞═════════╪═══════════╪════════════╡
│ 1       ┆ Jagger    ┆ Mick       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 2       ┆ O"Brian   ┆ Mary       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 3       ┆ Richards  ┆ Keith      │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 4       ┆ L"Etoile  ┆ Bennet     │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ ...     ┆ ...       ┆ ...        │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 5       ┆ Watts     ┆ Charlie    │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 6       ┆ Smith     ┆ D"Shawn    │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 7       ┆ Wyman     ┆ Bill       │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 8       ┆ Woods     ┆ Ron        │
├╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┤
│ 9       ┆ Jones     ┆ Brian      │
└─────────┴───────────┴────────────┘
```

And ~25% performance increase! :rocket: .

This will be on pypi on friday. Before that you can always compile from source, or use the suggestion made by @ghuls . Good luck!

**Comment by cbilot** [comment]
@ritchie46, Ok ... like wow ... I just parsed 392 million records in a csv file with only 6 fields ... in about 3 seconds.  It was so quick that I missed it.  I busted out laughing: "What's the sense of benchmarking **this**???  This is just _frighteningly_ fast."

FYI, I'm using the 0.10.5 version of Polars and Python 3.10.  Also, my csv file was in a RAM disk, so I/O is neglible.  The parsing pushed all 64 logical cores of my Threadripper Pro to nearly 100% - for a mere second or two.

@ghuls, thanks for the tip about xsv.  What an awesome tool!  I'm going to use this for double-checking total records and basic stats.
